Comments & Cleanup

yalegko opened the pull request:
Comment out everything which is public. Add links, examples and readme

yalegko committed:
Refactor and try to simplify tests code. Comment a bit

yalegko committed:
Document public API. Cleanup constants and code a bit

yalegko committed:
Cleanup and reorganize session.h a bit

yalegko committed:
Scratch README.md. Add short package doc

yalegko committed:
Drop the check of successful event sent in tests

We create lots of events in test code and sometimes this check become
flaky on CI at the moment of tearing down a test. We could probably
synchronize teardown of `generateEvents` routine, but it seems to be an
overkill (cos that place don't actually test our code)

mshsmlv commented:
Probably we need to add a mention about *Event usage in the callback, that EVENT_RECORD pointer isn't available after the callback.

yalegko committed:
Underline that Event could be used only inside EventCallback

Add a simple check that user tries to use *Event outside of a callback.
TODO: Consider adding some noCopy hack or even sync.Mutex to warn about
copies.

mshsmlv reviewed (approved):
LGTM
